src/etos_client/__main__.py

```diff
@@ -325,6 +325,7 @@ def main(args):
         success, results = test_result_handler.wait_for_test_suite_finished(spinner)
         if not success:
             spinner.fail(results)
+            sys.exit(not success)
```

Comment:
This will break a lot of things.

The `success` variable will be False if a single test case has failed within the sub suites, which will now result in ETOS Client exiting before downloading.
If we want to exit on Cancelled, then that must be done separately from test cases failing.

Reply:
Ah, seems I jumped the gun a bit here. Perhaps I could add some sort of reason-indication in the return tuple that we can condition the exit around, what do yo think?